src/Bedrock.Framework.Experimental/Protocols/Tls/BulkCiphers/BulkCipherKey.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Bedrock.Framework.Experimental.Protocols.Tls.BulkCiphers
+{
+    internal abstract class BulkCipherKey : IDisposable
+    {
+        public abstract Memory<byte> IV { get; }
```

Comment:
Why does this need to exist and why isn't it read only

Reply:
The IV is the initialisation vector that is used by the various symmetrical encryption algos. The spec defines how these are set so the "BulkCipher" will set them before every record that is encrypted (They contain a mixture of random/info from the handshake and counters). They are calculated differently for TLS 1.2 /1.3 etc and therefore cannot be controlled by the key implementation. They are mutated every record so they need to be editable.